### ISSUE TITLE
Quantization aware training loss reduce speed is very slower than the floating point training

### ISSUE THREAD

**Issue by mathmax12** [pytorch/pytorch#64155]
Hi 
 I found that if train with the same script and datasets the quantization aware training is very slow compared to the training with float data. To achieve the same loss error the QAT takes 1k epoch and the one using float point just needs around 400 epochs.

Note that both are training from scratch.

Is there a way that can help the QAT coverage s as fast as the float training?

cc @jerryzh168 @jianyuh @raghuramank100 @jamesr66a @vkuzo

**Comment by HDCharles** [comment]
Can you clarify what you mean exactly? It doesn't seem that its expected that those 2 operations will achieve comparable loss in the same amount of time/ever.

**Comment by mathmax12** [comment]
@HDCharles 
![image](https://user-images.githubusercontent.com/32367611/131448626-048cc8f7-a76b-493c-8b67-f333082df747.png)
![image](https://user-images.githubusercontent.com/32367611/131448641-decacdfb-7c5f-43b1-bc56-4f16c3c5756f.png)

The first image shows the sam loss change through the iterations.
You may notice that the loss decreasing for QAT is very slow compared to the loss.


Is this because the loss calculation and backpropagation both are floating for QAT and may help to accelerate QAT training is they are also quantized...(just some random thoughts)

**Comment by HDCharles** [comment]
the speed that things run can be sped up in a few ways but the optimization problem is still going to take more iterations to solve. You can use different optimizers and momentum to speed that up but in general its just a harder problem to solve. Its no longer locally continuous and neither is the derivative so the gradient updates tend to be much more sporadic as a result.
